pkg/server/server.go

```diff
@@ -518,6 +592,9 @@ func (s *Server) ListObjects(ctx context.Context, req *openfgav1.ListObjectsRequ
 			return nil, serverErrors.ValidationError(err)
 		}
 
+		if errors.Is(err, context.DeadlineExceeded) && result.ResolutionMetadata.WasThrottled.Load() {
+			return nil, serverErrors.ThrottledTimeout
```

Comment:
Do we have a testing this?

Reply:
As discussed, I have added a comment that currently this test is not feasible but leaving it here and adding comment to provide context on why this case wouldn't be possible. We can remove this logic if not required